text.go

```diff
@@ -13,6 +13,25 @@ const (
 	TextAlignTrailing
 )
 
+// TextTruncation controls how a `Label` or `Entry` will truncate its text.
+// The default value is `TextTruncateOff` which will not truncate.
+//
+// Since: 2.4
+type TextTruncation int
+
+const (
+	// TextTruncateOff means no truncation will be applied, it is the default.
+	// This means that the minimum size of a text block will be the space required to display it fully.
+	TextTruncateOff TextTruncation = iota
+	// TextTruncateClip will truncate text when it reaches the end of the available space.
+	TextTruncateClip
+	// TextTruncateEllipsis is like regular truncation except that an ellipses (…) will be inserted
+	// wherever text has been shortened to fit.
+	//
+	// Since: 2.4
+	TextTruncateEllipsis
```

Comment:
I remember there being an issue somewhere suggesting a middle ellipsis truncation mode eventually as well. Would it make more sense to name this one `TextTruncateTrailingEllipsis` then? (it is a bit of a mouthful) or is it ok to eventually have `TextTruncateEllipsis` and `TextTruncateMiddleEllipsis`?

Reply:
Good question. It depends on whether the leading/middle/trailing is to be separated from the ellipsis itself (some locales will want something other than an ellipsis).
I am guessing we will handle the rune in our localisation work so "ellipsis" is synonymous with the generic concept not the actual rune. In which case I think your suggestion is a good one.

That said, trailing vs leading depends on text direction - so I am pretty tempted to stick with current naming to mean "use ellipsis or similar, and truncate at the normal position for current direction" and we can add "middle ellipsis" in the future - or even something for the "other" end...